# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MIR.h
# Commit: cf96078694ae
# Full Hash: cf96078694aea74cae4557f683a76c6de6ba9cf1
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 3
# Description:
#   Bug 1692065 - wasm: Rename MIRType::RefOrNull to 'WasmAnyRef'. r=yury
#   
#   This is a mechanical change. The previous name 'RefOrNull' was too
#   generic, and it wasn't clear that the type was actually a wasm::AnyRef
#   type.
# ==============================================================================

diff -r 662cb63c42d7 -r cf96078694ae js/src/jit/MIR.h
--- a/js/src/jit/MIR.h	Fri Aug 04 14:06:49 2023 +0000
+++ b/js/src/jit/MIR.h	Fri Aug 04 14:06:49 2023 +0000
@@ -1468,7 +1468,7 @@
 
 class MWasmNullConstant : public MNullaryInstruction {
   explicit MWasmNullConstant() : MNullaryInstruction(classOpcode) {
-    setResultType(MIRType::RefOrNull);
+    setResultType(MIRType::WasmAnyRef);
     setMovable();
   }
 
@@ -2783,7 +2783,7 @@
     Compare_BigInt_String,
 
     // Wasm Ref/AnyRef/NullRef compared to Ref/AnyRef/NullRef
-    Compare_RefOrNull,
+    Compare_WasmAnyRef,
   };
 
  private:
@@ -2818,7 +2818,7 @@
                compareType == Compare_Int64 || compareType == Compare_UInt64 ||
                compareType == Compare_Double ||
                compareType == Compare_Float32 ||
-               compareType == Compare_RefOrNull);
+               compareType == Compare_WasmAnyRef);
     auto* ins = MCompare::New(alloc, left, right, jsop, compareType);
     ins->setResultType(MIRType::Int32);
     return ins;
@@ -2902,7 +2902,7 @@
       case Compare_Int64:
       case Compare_UInt64:
       case Compare_UIntPtr:
-      case Compare_RefOrNull:
+      case Compare_WasmAnyRef:
         return false;
     }
     MOZ_CRASH("unexpected compare type");
@@ -2960,8 +2960,8 @@
       case Compare_BigInt_String:
         ty = "BigInt_String";
         break;
-      case Compare_RefOrNull:
-        ty = "RefOrNull";
+      case Compare_WasmAnyRef:
+        ty = "WasmAnyRef";
         break;
       default:
         ty = "!!unknown!!";
@@ -9366,7 +9366,7 @@
 
     // The only types supported at the moment.
     MOZ_ASSERT(type == MIRType::Pointer || type == MIRType::Int32 ||
-               type == MIRType::Int64 || type == MIRType::RefOrNull);
+               type == MIRType::Int64 || type == MIRType::WasmAnyRef);
 
     setMovable();
     setResultType(type);
@@ -9409,7 +9409,7 @@
 
     // The only types supported at the moment.
     MOZ_ASSERT(type == MIRType::Pointer || type == MIRType::Int32 ||
-               type == MIRType::Int64 || type == MIRType::RefOrNull);
+               type == MIRType::Int64 || type == MIRType::WasmAnyRef);
   }
 
  public:
@@ -9940,7 +9940,7 @@
         instanceDataOffset_(instanceDataOffset),
         isConstant_(isConstant) {
     MOZ_ASSERT(IsNumberType(type) || type == MIRType::Simd128 ||
-               type == MIRType::Pointer || type == MIRType::RefOrNull);
+               type == MIRType::Pointer || type == MIRType::WasmAnyRef);
     setResultType(type);
     setMovable();
   }
@@ -9995,7 +9995,7 @@
                               public NoTypePolicy::Data {
   MWasmLoadTableElement(MDefinition* elements, MDefinition* index)
       : MBinaryInstruction(classOpcode, elements, index) {
-    setResultType(MIRType::RefOrNull);
+    setResultType(MIRType::WasmAnyRef);
     setMovable();
   }
 
@@ -10171,7 +10171,7 @@
     MOZ_ASSERT(valueOffset <= INT32_MAX);
     MOZ_ASSERT(valueBase->type() == MIRType::Pointer ||
                valueBase->type() == MIRType::StackResults);
-    MOZ_ASSERT(value->type() == MIRType::RefOrNull);
+    MOZ_ASSERT(value->type() == MIRType::WasmAnyRef);
     initOperand(0, instance);
     initOperand(1, valueBase);
     initOperand(2, value);
@@ -11325,7 +11325,7 @@
         aliases_(aliases),
         maybeTrap_(maybeTrap) {
     MOZ_ASSERT(offset <= INT32_MAX);
-    MOZ_ASSERT(value->type() != MIRType::RefOrNull);
+    MOZ_ASSERT(value->type() != MIRType::WasmAnyRef);
     // "if you want to narrow the value when it is stored, the source type
     // must be Int32".
     MOZ_ASSERT_IF(narrowingOp != MNarrowingOp::None,
@@ -11388,9 +11388,9 @@
         preBarrierKind_(preBarrierKind) {
     MOZ_ASSERT(obj->type() == TargetWordMIRType() ||
                obj->type() == MIRType::Pointer ||
-               obj->type() == MIRType::RefOrNull);
+               obj->type() == MIRType::WasmAnyRef);
     MOZ_ASSERT(offset <= INT32_MAX);
-    MOZ_ASSERT(value->type() == MIRType::RefOrNull);
+    MOZ_ASSERT(value->type() == MIRType::WasmAnyRef);
     MOZ_ASSERT(
         aliases.flags() ==
             AliasSet::Store(AliasSet::WasmStructInlineDataArea).flags() ||